pennylane/circuit_graph.py

```diff
@@ -324,7 +331,13 @@ def ancestors(self, ops):
         Returns:
             set[Operator]: ancestors of the given operators
         """
-        return set().union(*(nx.dag.ancestors(self._graph, o) for o in ops)) - set(ops)
+        anc = set(
+            self._graph.get_node_data(n)
+            for n in set().union(
+                *(rx.ancestors(self._graph, self._graph.nodes().index(o)) for o in ops)
+            )
```

Comment:
Could you explain the additional for loop here (for my own understanding!) 🙂

Reply:
This is a good one! :slightly_smiling_face: the `ancestors` method in RX returns node indexes instead of node-values! So, after computing the union of ancestors of `o` in `ops`, the additional for loop fetches the node-values.

Reply:
thanks @maliasadi for explaining! as above might be good to leave a code comment